proof/crefine/RISCV64/ADT_C.thy

```diff
@@ -1176,62 +1176,103 @@ lemma scRefills_unique:
   done
 
 lemma scSize_unique:
-  assumes "refill_buffer_relation ah ch gs" "refill_buffer_relation ah' ch gs"
-      and "ksPSpace s = ah \<and> ksPSpace s p = Some (KOSchedContext sc)"
-      and "ksPSpace s' = ah' \<and> ksPSpace s' p = Some (KOSchedContext sc')"
+  assumes "refill_buffer_relation (ksPSpace as) ch gs" "refill_buffer_relation (ksPSpace as') ch gs"
+      and "ksPSpace as p = Some (KOSchedContext sc)"
+      and "ksPSpace as' p = Some (KOSchedContext sc')"
```

Comment:
It looks like `ksSpace` is not needed here. Could it be just the following?

```suggestion
  assumes "refill_buffer_relation ah ch gs" "refill_buffer_relation ah' ch gs"
      and "ah p = Some (KOSchedContext sc)"
      and "ah' p = Some (KOSchedContext sc')"
```

Reply:
Yep, done, thanks